addons/spreadsheet/static/src/chart/plugins/odoo_chart_ui_plugin.js

```diff
@@ -131,6 +135,51 @@ export class OdooChartUIPlugin extends OdooUIPlugin {
         return this.charts[dataSourceId];
     }
 
+    /**
+     * Get the callback used for onClick and onHover in an Odoo Chart
+     */
+    getChartDatasetActionCallbacks(datasets, labels, resModel, actionXmlId) {
```

Comment:
I would have given only the chartId here.
And get everything needed here. The simpler API would decouple the getter implementation from the call site 
```suggestion
    getChartDatasetActionCallbacks(chartId) {
```